osu.Desktop/OsuGameDesktop.cs

```diff
@@ -54,19 +56,29 @@ public override void SetHost(GameHost host)
         private void dragDrop(DragEventArgs e)
         {
             // this method will only be executed if e.Effect in dragEnter gets set to something other that None.
-            var dropData = e.Data.GetData(DataFormats.FileDrop) as object[];
+            var dropData = (object[])e.Data.GetData(DataFormats.FileDrop);
             var filePaths = dropData.Select(f => f.ToString()).ToArray();
-            ImportBeatmapsAsync(filePaths);
+
+            if (filePaths.All(f => Path.GetExtension(f) == @".osz"))
+                Task.Run(() => BeatmapDatabase.Import(filePaths));

```

Comment:
Thing is, you already show the icon for a valid drag when dragging in a mixed list of osz and osr. Both functions should be consistent, so no valid drag should be shown in that case.